Grant Details Page - move grants.gov, copy, print buttons

amyewang opened the pull request:
### Ticket #3153
## Description
Update the Grant Details page main actions buttons to match expected state in the [Figma designs](https://www.figma.com/design/L2gfVMBH0x1daOlOe50sJl/Follow-%2F-Leave-a-Note?node-id=1101-13791&t=Co6m063mfZx7NQ1c-0)  
Moved to the right column, updated text and icons, updated "copy" and "print" to be links. 

## Screenshots / Demo Video
<img width="1326" alt="image" src="https://github.com/usdigitalresponse/usdr-gost/assets/13317264/e0989b25-6f90-4916-bafd-ad3c66482152">

## Testing
Tested various window widths in local dev env 

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [ ] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [ ] Added PR reviewers

jeffsmohan commented on packages/client/src/views/GrantDetailsView.vue:
```diff
@@ -29,6 +29,65 @@
+                role="button"
+                :variant="copyUrlSuccessTimeout === null ? 'outline-primary' : 'outline-success'"
+                data-dd-action-name="copy btn"
+                style="white-space: nowrap; "
```
In general, we should avoid inline styles where possible, since they're harder to maintain, keep consistent, override, etc. Instead, we want to favor existing [utility classes](https://bootstrap-vue.org/docs/reference/utility-classes) where available, or styles defined in a `style` section of the component otherwise. 

In this case, you should be able to use the `text-nowrap` utility class provided by Bootstrap. ([Docs here](https://getbootstrap.com/docs/4.5/utilities/text/#text-wrapping-and-overflow))

amyewang replied:
thanks! getting used to Bootstrap

jeffsmohan replied:
Sigh... me too... :P

jeffsmohan commented on packages/client/src/views/GrantDetailsView.vue:
```diff
@@ -29,6 +29,65 @@
+              variant="primary"
+              block
+              class="mr-5"
+              style="width: fit-content; white-space: nowrap; "
```
A cleaner way to achieve the new layout, using existing bootstrap helpers: 

1. Remove the `block` option (instead of the `width: fit-content` style)
2. Add the `text-nowrap` utility class (instead of the `white-space: nowrap` style)
3. You can remove the `mr-5` class now that we don't need a right margin on the button

amyewang replied:
<img width="400" alt="image" src="https://github.com/usdigitalresponse/usdr-gost/assets/13317264/14785a3d-405d-4ed8-a266-f9e9d28d6750">

I left a margin to the right of the button to add some negative space between "view" and "copy" in the case of the minimum viewport width - this wasn't really specified in the designs though. Let me know what you think

jeffsmohan replied:
Good call! Sorry, I just assumed it had been carried over without intention from the previous arrangement, my bad. 

Longer-term, I know the team is looking at supporting mobile designs on smaller screen sizes, so we might be able to address this sort of thing even more robustly...

jeffsmohan commented on packages/client/src/views/GrantDetailsView.vue:
```diff
@@ -29,6 +29,65 @@
+                role="button"
+                variant="outline-primary"
+                data-dd-action-name="print btn"
+                style="white-space: nowrap; "
```
Same as above

jeffsmohan commented on packages/client/src/views/GrantDetailsView.vue:
```diff
@@ -29,6 +29,65 @@
+                <span v-if="copyUrlSuccessTimeout === null">Copy Link</span>
+                <span v-else>Link Copied</span>
+              </a>
+              <div class="col-1 border-right border-dark p-2"/>
```
Seeing some linting errors here. Ideally you shouldn't have to think about style stuff like this, and you can get your editor set up to auto-format on save. A bit of documentation on getting set up is here — https://github.com/usdigitalresponse/usdr-gost/blob/main/docs/development.md#linting — but lmk if you have any trouble getting that working!

jeffsmohan reviewed (changes requested):
This is looking really good! Just a few minor cleanups in inline comments, plus: 

- Might be worth checking with the designers, but to my eye I think the designs had the button going down a size smaller, plus the margin between the button/links and the rest of the content being a bit smaller as well

amyewang commented:
Thanks for the review @jeffsmohan ! I fixed the inline styling, updated the overall button & link sizing (thanks for that callout), and updated my linting. Left 1 small comment about the button margin.

jeffsmohan commented on usdr-gost.code-workspace:
```diff
@@ -26,6 +26,6 @@
 		// to manage linting/fixing, but users can have other settings via "formatOnSave".
 		// If both are on, the editor can "fix" things that are immediately flagged as lint failures.
-		"editor.formatOnSave": false,
+		"editor.formatOnSave": true,
```
I don't think we want to check this change in. Were you not able to get automatic linting on save working without this change? 

I'd love to look into that a bit more with you... the `codeActionsOnSave` is supposed to supersede the `formatOnSave` setup in VSCode I believe, but the setup is finicky, and I didn't have many users to test out the settings with when I was changing them last...

amyewang replied:
whoops sorry this was careless, I was playing with settings and accidentally committed this. my bad.

jeffsmohan reviewed (changes requested):
This looks basically ready to go! 

Only thing is let's avoid changing the workspace settings, at least in this PR. I'd love to discuss what linting issues you were running into that this solved for you, and then maybe we can turn that into a separate PR to get linting set up more robustly for folks!

jeffsmohan reviewed (approved):
lgtm!
